avocado/core/loader.py

```diff
@@ -613,7 +613,7 @@ def _find_avocado_tests(self, path):
         # The name used, in case of 'import avocado as avocadolib'
         mod_import_name = None
         # The resulting test classes
-        result = {}
+        result = collections.OrderedDict()
```

Comment:
There's an order already. So the mentioned danger is already in place. But that order is not easily predictable without listing the tests, since it depends on the python dictionary ordering based on the hash of the elements. I'm proposing to take control of that order. First method that came to my mind was the same method used to order the test methods from a test class: discovery order. I'd be ok with alphabetical order, but that would not change the current state that much. So, let me drop this commit as it's not popular and not really relevant for this PR. We can discuss that later.